### PR TITLE
builtin: add make_set built-in function

### PR DESCRIPTION
**Pull request by ariesdevil**

### CODE REVIEW

File: expression/builtin_string.go [hunk 1]
```diff
@@ -1395,7 +1397,41 @@ type builtinMakeSetSig struct {
 
 // See https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_make-set
 func (b *builtinMakeSetSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("make_set")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return types.Datum{}, errors.Trace(err)
+	}
+	d.SetString("")
```
Comment (shenli) [inline expression/builtin_string.go]: if args[0].IsNull(), mysql will return null.
mysql> SELECT MAKE_SET(NULL,'a','b','c');
+----------------------------+
| MAKE_SET(NULL,'a','b','c') |
+----------------------------+
| NULL                       |
+----------------------------+

File: expression/builtin_string.go [hunk 2]
```diff
@@ -1395,7 +1397,41 @@ type builtinMakeSetSig struct {
...
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	bitsValue := stringutil.Reverse(strconv.FormatInt(arg0, 2))
+
+	for i := 1; i < len(args) && i-1 < len(bitsValue); i++ {
+		if args[i].IsNull() {
+			continue
+		}
+		if string(bitsValue[i-1]) == "1" {
```
Comment (shenli) [inline expression/builtin_string.go]: It is better to use int bit-wise and shift operation. Not string compare.

**Comment by ariesdevil** [comment]
@shenli fixed and plz review again.

**Comment by coocood** [comment]
LGTM
Thank you for your PR!

**Comment by coocood** [comment]
Sorry for the delay, there are many PRs conflicting on typeinfer files.
Merging each one makes other PRs conflicting, so we have to queue the PRs.
 when this PR is ready to merge, we will notify you to resolve conflicts.

**Comment by ariesdevil** [comment]
OK, got it.

**Comment by tiancaiamao** [comment]
The test could not reflect the reality, if you run the binary, you'll get this:

```
mysql> SELECT MAKE_SET(-100 | 4,'a','b','c');
ERROR 1105 (HY000): constant 18446744073709551516 overflows bigint
```

That's because, `-100 | 4` will be calculate directly in test code, while it should be done in TiDB as an expression.
  
```
{[]interface{}{-100 | 4, "hello", "nice", "abc", "world"}, "abc,world"},
```

Okay, it involves another problem.

LGTM

**Comment by ariesdevil** [comment]
@tiancaiamao I test on my machine and it works well (mysql version: 5.7.17):

```
mysql> SELECT MAKE_SET(-100 | 4,'a','b','c');
+--------------------------------+
| MAKE_SET(-100 | 4,'a','b','c') |
+--------------------------------+
| c                              |
+--------------------------------+
1 row in set (0.00 sec)
```

**Pull request merged**